File: akshat/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        s_dict = {i: s.count(i) for i in set(s)} #hot one liners 
        t_dict = {i: t.count(i) for i in set(t)}
        if s_dict == t_dict:
            return True
        else:
            return False
        

File: gabe/valid_anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char, int> mapping;

        for (char c : s) {
            mapping[c]++;
        }

        for (char c: t) {
            mapping[c]--;
        }

        for(auto const& [key, val]: mapping) {
            if (val != 0) return false;
        }
        return true;
    }
};